modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/ior.kt

```diff
@@ -169,3 +176,46 @@ interface IorHash<L, R> : Hash<Ior<L, R>>, IorEq<L, R> {
 
 fun <L, R> Ior.Companion.fx(SL: Semigroup<L>, c: suspend MonadSyntax<IorPartialOf<L>>.() -> R): Ior<L, R> =
   Ior.monad(SL).fx.monad(c).fix()
+
+@extension
+interface IorEqK<L> : EqK<IorPartialOf<L>> {
+  fun EQL(): Eq<L>
+
+  override fun <A> Kind<IorPartialOf<L>, A>.eqK(other: Kind<IorPartialOf<L>, A>, EQ: Eq<A>): Boolean =
+    (this.fix() to other.fix()).let { (ls, rs) ->
+      Ior.eq(EQL(), EQ).run { ls.eqv(rs) }
+    }
+}
+
+@extension
+interface IorCrosswalk<L> : Crosswalk<IorPartialOf<L>>, IorFunctor<L>, IorFoldable<L> {
+  override fun <F, A, B> crosswalk(ALIGN: Align<F>, a: Kind<IorPartialOf<L>, A>, fa: (A) -> Kind<F, B>): Kind<F, Kind<IorPartialOf<L>, B>> {
+    return when (val ior = a.fix()) {
+      is Ior.Left -> ALIGN.run { empty<Kind<IorPartialOf<L>, B>>() }
+      is Ior.Both -> ALIGN.run { fa(ior.rightValue).map { Ior.Both(ior.leftValue, it) } }
+      is Ior.Right -> ALIGN.run { fa(ior.value).map { it.rightIor() } }
+    }
+  }
+}
+
+interface IorBicrosswalk : Bicrosswalk<ForIor>, IorBifunctor, IorBifoldable {
+  override fun <F, A, B, C, D> bicrosswalk(
+    ALIGN: Align<F>,
+    tab: Kind2<ForIor, A, B>,
+    fa: (A) -> Kind<F, C>,
+    fb: (B) -> Kind<F, D>
+  ): Kind<F, Kind2<ForIor, C, D>> =
+    when (val e = tab.fix()) {
+      is Ior.Left -> ALIGN.run {
+        fa(e.value).map { it.leftIor() }
+      }
+      is Ior.Right -> ALIGN.run {
+        fb(e.value).map { it.rightIor() }
+      }
+      is Ior.Both -> ALIGN.run {
+        align(fa(e.leftValue), fb(e.rightValue))
+      }
+    }
+}
+
+fun Ior.Companion.bicrosswalk() = object : IorBicrosswalk {}
```

Comment:
`@extension`? Ignore this if it does not work, but if it does it's usually much better as it generates methods as well.

Reply:
same